### PR TITLE
Us107 edit ce details

### PR DESCRIPTION
**Pull request by emilioward**
This PR adds functionality to allow an internal user to update the `shown to respondent as date` in a collection exercise.

Run this PR with the PR for response operations UI to test changes

https://trello.com/c/a3s4Yy3x

### CODE REVIEW

**Comment by emilioward** [comment]
[US107: INT Edit Collection Exercise details](https://trello.com/c/a3s4Yy3x/61-us107-int-user-to-edit-amend-ce)

File: test/resources/test_collection_exercise.py [hunk 1]
```diff
@@ -468,3 +474,30 @@ def test_single_collection_exercise_classifier_type_fail(self, mock_request):
...
+    def test_update_collection_exercise_details_fail(self, mock_request):
+        mock_request.put(url_update_ce_details, status_code=500)
+        url = f'backstage-api/v1/collection-exercise/update-collection-exercise-details/{collection_exercise_id}'
+        response = self.app.put(url, headers=self.headers, data=json.dumps({
+            "user_description": '201901'
+        }))
+        self.assertEqual(response.status_code, 500)
+
+    @requests_mock.mock()
+    def test_update_collection_exercise_details_no_found(self, mock_request):
```
Comment (gemmai95) [inline test/resources/test_collection_exercise.py]: Bit of a minor thing but can you change this to not found rather than no found?

**Comment by jcox-dev** [comment]
Works for me  👍 

image

File: ras_backstage/controllers/collection_exercise_controller.py [hunk 2]
```diff
@@ -127,3 +127,20 @@ def execute_collection_exercise(collection_exercise_id):
         raise ApiError(url, response.status_code)
 
     logger.debug('Successfully executed collection exercise', collection_exercise_id=collection_exercise_id)
+
+
+def update_collection_exercise_user_description(collection_exercise_id, user_description):
+    logger.debug('Updating collection exercise', collection_exercise_id=collection_exercise_id,
```
Comment (tej99) [inline ras_backstage/controllers/collection_exercise_controller.py]: Is it worth being more specific in the log? 'Updating collection exercise user description

File: test/resources/test_collection_exercise.py [hunk 3]
```diff
@@ -64,6 +66,10 @@ class TestCollectionExercise(unittest.TestCase):
 
     def setUp(self):
         self.app = app.test_client()
+        self.headers = {
+            'Authorization': 'test_jwt',
```
Comment (tej99) [inline test/resources/test_collection_exercise.py]: Authorization should not be required in the headers for these tests

File: test/resources/test_collection_exercise.py [hunk 4]
```diff
@@ -468,3 +474,30 @@ def test_single_collection_exercise_classifier_type_fail(self, mock_request):
         # Then
         self.assertEqual(response.status_code, 500)
         self.assertEqual(response_data['error']['status_code'], 500)
+
+    @requests_mock.mock()
```
Comment (tej99) [inline test/resources/test_collection_exercise.py]: Missing test case, bad request when required field (user_description) is not given in the payload.

File: ras_backstage/controllers/collection_exercise_controller.py [hunk 5]
```diff
@@ -127,3 +127,38 @@ def execute_collection_exercise(collection_exercise_id):
...
+        raise ApiError(url, response.status_code)
+    if response.status_code not in (200, 201, 202):
+        logger.error('Error updating collection exercise user description', collection_exercise_id=collection_exercise_id)
+        raise ApiError(url, response.status_code)
+
+    logger.debug('Successfully updated collection exercise user description',
+                 collection_exercise_id=collection_exercise_id)
+
+
+def update_collection_exercise_period(collection_exercise_id, period):
```
Comment (jcox-dev) [inline ras_backstage/controllers/collection_exercise_controller.py]: Missing test cases for period.

**Pull request merged**